awx/ui/src/components/LaunchPrompt/LaunchPrompt.js

```diff
@@ -53,6 +62,61 @@ function PromptModalForm({
     setValue('extra_vars', mergeExtraVars(extraVars, surveyValues));
     setValue('scm_branch', values.scm_branch);
     setValue('verbosity', values.verbosity);
+    setValue('timeout', values.timeout);
+    setValue('forks', values.forks);
+    setValue('job_slice_count', values.job_slice_count);
+    setValue('execution_environment', values.execution_environment?.id);
+
+    if (launchConfig.ask_instance_groups_on_launch) {
+      const instanceGroupIds = [];
+      values.instance_groups.forEach((instance_group) => {
+        instanceGroupIds.push(instance_group.id);
+      });
+      setValue('instance_groups', instanceGroupIds);
+    }
+
+    if (launchConfig.ask_labels_on_launch) {
+      const labelIds = [];
+      const newLabels = [];
+      const labelRequests = [];
+      let organizationId = resource.organization;
+      values.labels.forEach((label) => {
+        if (typeof label.id !== 'number') {
+          newLabels.push(label);
+        } else {
+          labelIds.push(label.id);
+        }
+      });
+
+      if (newLabels.length > 0) {
+        if (!organizationId) {
+          // eslint-disable-next-line no-useless-catch
+          try {
+            const {
+              data: { results },
+            } = await OrganizationsAPI.read();
+            organizationId = results[0].id;
+          } catch (err) {
+            throw err;
+          }
+        }
+      }
+
+      newLabels.forEach((label) => {
+        labelRequests.push(
+          LabelsAPI.create({
+            name: label.name,
+            organization: organizationId,
+          }).then(({ data }) => {
+            labelIds.push(data.id);
+          })
+        );
+      });
+
+      await Promise.all(labelRequests);
+
+      setValue('labels', labelIds);
```

Comment:
Consider moving most or all of this into a helper function to keep `handleSubmit` down to a scanable length

Reply:
Good call - there's identical code in the Schedule Edit component so I'll pull this all out and put it somewhere where it can be imported.